lightning-invoice/src/payment.rs

```diff
@@ -1397,18 +1564,263 @@ mod tests {
+			.expect_send(Amount::OnRetry(final_value_msat / 2));
+		let final_value_msat = payment_invoice.amount_milli_satoshis().unwrap();
+		let router = TestRouter {};
+		let scorer = RefCell::new(TestScorer::new()
+			// 1st invoice, 1st path
+			.expect_usage(ChannelUsage { amount_msat: 10, inflight_htlc_msat: 0, effective_capacity: EffectiveCapacity::Unknown } )
+			.expect_usage(ChannelUsage { amount_msat: 20, inflight_htlc_msat: 0, effective_capacity: EffectiveCapacity::Unknown } )
+			.expect_usage(ChannelUsage { amount_msat: 64, inflight_htlc_msat: 0, effective_capacity: EffectiveCapacity::Unknown } )
+			// 1st invoice, 2nd path
+			.expect_usage(ChannelUsage { amount_msat: 64, inflight_htlc_msat: 0, effective_capacity: EffectiveCapacity::Unknown } )
+			// Retry 1
+			.expect_usage(ChannelUsage { amount_msat: 10, inflight_htlc_msat: 0, effective_capacity: EffectiveCapacity::Unknown } )
+			.expect_usage(ChannelUsage { amount_msat: 20, inflight_htlc_msat: 64, effective_capacity: EffectiveCapacity::Unknown } )
+			.expect_usage(ChannelUsage { amount_msat: 32, inflight_htlc_msat: 0, effective_capacity: EffectiveCapacity::Unknown } )
+			.expect_usage(ChannelUsage { amount_msat: 32, inflight_htlc_msat: 64, effective_capacity: EffectiveCapacity::Unknown } )
+			// Retry 2
+			.expect_usage(ChannelUsage { amount_msat: 10, inflight_htlc_msat: 0, effective_capacity: EffectiveCapacity::Unknown } )
+			.expect_usage(ChannelUsage { amount_msat: 20, inflight_htlc_msat: 96, effective_capacity: EffectiveCapacity::Unknown } )
+			.expect_usage(ChannelUsage { amount_msat: 32, inflight_htlc_msat: 0, effective_capacity: EffectiveCapacity::Unknown } )
+			.expect_usage(ChannelUsage { amount_msat: 32, inflight_htlc_msat: 96, effective_capacity: EffectiveCapacity::Unknown } )
+		);
+		let logger = TestLogger::new();
+		let invoice_payer =
+			InvoicePayer::new(&payer, router, &scorer, &logger, event_handler, Retry::Attempts(2));
+
+		// Fail 1st path, leave 2nd path inflight
+		let payment_id = Some(invoice_payer.pay_invoice(&payment_invoice).unwrap());
+		invoice_payer.handle_event(&Event::PaymentPathFailed {
+			payment_id,
+			payment_hash,
+			network_update: None,
+			rejected_by_dest: false,
+			all_paths_failed: false,
+			path: TestRouter::path_for_value(final_value_msat),
+			short_channel_id: None,
+			retry: Some(TestRouter::retry_for_invoice(&payment_invoice)),
+		});
+
+		// Fails again the 1st path of our retry
+		invoice_payer.handle_event(&Event::PaymentPathFailed {
+			payment_id,
+			payment_hash,
+			network_update: None,
+			rejected_by_dest: false,
+			all_paths_failed: false,
+			path: TestRouter::path_for_value(final_value_msat / 2),
+			short_channel_id: None,
+			retry: Some(RouteParameters {
+				final_value_msat: final_value_msat / 2,
+				..TestRouter::retry_for_invoice(&payment_invoice)
+			}),
+		});
+	}
+
+	#[test]
+	fn accounts_for_all_inflight_htlcs_sent_during_partial_failure() {
```

Comment:
This test should be "some" and the other test should be "all" since `ChannelUnavailable` doesn't result in anything sent.

Reply:
oh, you're right. oops.